zio-cli/shared/src/main/scala/zio/cli/FigFontParser.scala

```diff
@@ -0,0 +1,336 @@
+package zio.cli
+
+import scala.collection.mutable
+
+// FigLettering Font (.flf), see:
+// http://www.jave.de/figlet/figfont.html or
+// https://github.com/cmatsuoka/figlet/blob/master/figfont.txt
+final case class FigFont(
+  header: FigHeader,
+  rightToLeft: Boolean,
+  layout: Layout,
+  chars: Map[Char, FigChar]
+)
+
+final case class FigChar(lines: Seq[FigCharLine], width: Int, height: Int)
```

Comment:
Rather than use the type `Seq`, we should probably use `zio.Chunk` for high-performance and immutability.

Reply:
Initially, i was using `Chunk` everywhere there is `Seq` now, but for some reason was getting `NoSuchMethodException` in `Chunk.head` -- i wonder if the issue was caused by the parser being in the shared project while the tests in the jvm project -- I'll try to restore `Chunk`.

Reply:
Try `chunk(0)` instead of `chunk.head` (although keep in mind it is unsafe).